packages/driver/src/cy/commands/screenshot.js

```diff
@@ -347,6 +347,16 @@ const takeScreenshot = (Cypress, state, screenshotConfig, options = {}) => {
 
   return before()
   .then(() => {
+    const win = state('window')
+
+    // in the case that an element might change size on scroll
+    // we trigger a scroll event to ensure that all elements are
+    // at their final size before we calculate the total height
+    // since we scroll down the page in takeScrollingScreenshots
+    // and don't want the page size to change once we start
+    // https://github.com/cypress-io/cypress/issues/6099
+    win.dispatchEvent(new CustomEvent('scroll'))
```

Comment:
```suggestion
    win.dispatchEvent(new win.CustomEvent('scroll'))
```

This might fix the cross-origin error that's failing the e2e tests.